src/index.js

```diff
@@ -949,28 +1064,39 @@ export default function (babel) {
             " with a (key, value) pair.");
           }
 
+          const node = tailPath.node;
+          const n = (...args) => nodeAt(node, ...args);
           const keyExpr = seqExpr.expressions[0];
           const valExpr = seqExpr.expressions[1];
 
-          return t.assignmentExpression(
+          return n("assignmentExpression",
             "=",
-            t.memberExpression(id, keyExpr, true),
+            n("memberExpression", id, keyExpr, true),
             valExpr
           );
         });
 
-        path.replaceWith(wrapComprehensionInIife(id, t.objectExpression([]), path.node.loop));
+        path.replaceWith(
+          wrapComprehensionInIife(
+            id,
+            nodeAt(path.node, "objectExpression", []),
+            path.node.loop
+          )
+        );
       },
 
       TildeCallExpression: {
         // run on exit instead of enter so that SafeMemberExpression
         // can process differently from a wrapping CallExpression
         // eg; `a?.b~c()` -> `a == null ? null : c(a.b)`
         exit(path) {
-          const callExpr = t.callExpression(path.node.right, [
-            path.node.left,
-            ...path.node.arguments,
-          ]);
+          // XXX: this might be breaking source maps because it makes impossible
+          // ordering, right appears after left in original source but before
```

Comment:
I'm starting to think that "impossible ordering" might really be fine in many cases, actually... this being a great example. 

If there was an error in 
```
a.foo(x.y)
```
which came from
```
x.y~a.foo()
```
due to, say, `foo` not being a function, 

you'd want the sourcemap for the error to point like this:
```
x.y~a.foo()
      ^^^
```

even though in the actual code, you'd have "impossible positions"

in general, thinking about, "if an error occurred at the node we're generating, where in the lsc source would we want to point the user?" might be the best approach... having a hard time thinking about how to make that thought-process easier through tooling though... 

thoughts?